### PR TITLE
Extract coercion

### PR DESCRIPTION
**Pull request by joel**
Add for prevent usage of both option :parse and :type

### CODE REVIEW

File: lib/csv_row_model/coercer.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    def decode(value)
+      caller.instance_exec(value, &parse_lambda)
+    end
+
+    private
+
+    # @return [Lambda, Proc] returns the Lambda/Proc given in the parse option or:
+    # ->(original_value) { parse_proc_exists? ? parsed_value : original_value  }
+    def parse_lambda
+      raise ArgumentError.new("You need either :parse OR :type but not both of them") if options[:parse] && options[:type]
```
Comment (s12chung) [inline lib/csv_row_model/coercer.rb:13]: i like this.

**Comment by s12chung** [comment]
i generally dislike files that have one function, unless they're shared.

if you feel strongly about separating it, go ahead. but if you're testing that file, i believe it deserves it's own `_spec` file.

**Comment by joel** [comment]
For me it's easier to find out what it expected here when it have a file for one function #SingleRresponsibilityPrinciple this my point of view :wink:

**Comment by s12chung** [comment]
:p.... i just watched a talk yesterday about that.... i think it's the principle i dislike the most, because it conflicts with "tell don't ask" and adds another layer of indirection.

**Comment by joel** [comment]
@s12chung can you give me the link of your talk? Because I can't see why this can be conflict, and I remember comment of you when I applied 'tell don't ask' pattern on the core ;-)

**Pull request merged**